resources/analysisTools/copyNumberEstimationWorkflow/beagle_embed_haplotypes_vcf.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python
+import argparse
+import gzip
+
+def is_comment_line(line):
+    return line.startswith("#")
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--hap_file', help='Input file with phased haplotypes (i.e. output from Beagle)')
+parser.add_argument('--vcf_file', help='Input .vcf file')
+parser.add_argument('--out_file', help='Out .vcf file')
+args = parser.parse_args()
+
+if args.hap_file.split('.')[-1] == 'gz':
+    hap_file = gzip.open( args.hap_file, "r" )
+else:
+    hap_file = open( args.hap_file, "r" )
+vcf_file = open( args.vcf_file, "r" )
+outfile = open( args.out_file, "w" )
+
+hap_line = hap_file.readline()
+while is_comment_line(hap_line):
+    hap_line = hap_file.readline()
+
+if hap_line:
+    hap_line = hap_line.rstrip().split()
+
+for vcf_line in vcf_file:
+    
+    if not is_comment_line(vcf_line):
+        
+        vcf_line = vcf_line.rstrip().split("\t")
+        vcf_line[0] = vcf_line[0].replace('chr', '')
```

Comment:
These columns of vcf_line, do they have descriptive names? Maybe declare variables with names at the top and use the names for accessing the fields?

```
Chromosome = 0
...
vcf_line[Chromosome]
...
```

Reply:
Though columns 0, 8 have descriptive names, column 9 doesn't have one and will be path to the BAM file. These are default VCF columns, so the order will not change. I will leave it like this now, else it will look half-worked.